app/controllers/users_controller.rb

```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "Register success"
+      redirect_to root_path
+    else
+      flash[:success] = "Register failed"
```

Comment:
Em đọc về cách dùng flash ở đây https://guides.rubyonrails.org/action_controller_overview.html#the-flash
Theo đó thì khi `render :new` em không thể dùng `flash[:success] = ...` dc